tests/test_pricing_model.py

```diff
@@ -0,0 +1,1281 @@
+                    # We set up the problem as:
+                    #   (100_000 - d_z) ** (1 - T) + 300_100 ** (1 - T) = k
+                    #
+                    # Solving for d_z, we get the following calculation:
+                    #   d_z = 100_000 - (k - 300_100 ** (1 - T)) ** (1 / (1 - T)) = 97.55314236719278
+                    #
+                    # The output is d_x = c * d_z. Since c = 1, d_x = d_z.
+                    # Note that this is slightly smaller than the without slippage value
+                    without_fee=97.55314236719278,
+                    # Combining the without_fee and the fee, we calculate with_fee as:
+                    #   with_fee = 97.55314236719278 - 0.48908371610497 = 97.0640586510878
+                    with_fee=97.0640586510878,
+                ),
+            ),
+            # Medium slippage trade - in_ is 10% of share reserves.
+            (
+                TestCaseCalcOutGivenInSuccess(
+                    in_=10_000,
+                    share_reserves=100_000,
+                    bond_reserves=100_000,
+                    token_out="base",
+                    fee_percent=0.01,
+                    days_remaining=182.5,
+                    time_stretch_apy=0.05,
+                    share_price=1,
+                    init_share_price=1,
+                ),
+                # From the input, we have the following values:
+                # - T = 0.02253584403159705
+                # - p = 1.0250671833648672
+                # - k = 302_929.51067963685
+                TradeResult(
+                    # Using the spot price, the expected output without slippage or fees is given by:
+                    #   (1 / 1.0250671833648672) * 10_000 = 9755.458141947514
+                    without_fee_or_slippage=9755.458141947514,
+                    # Since we are buying bonds, in_ is an amount of base and we calculate the fee using the spot price as:
+                    #   fee = 0.01 * (1 - (1 / p)) * 10_000 = 2.4454185805248496
+                    fee=2.4454185805248496,
+                    # We set up the problem as:
+                    #   (100_000 - d_z) ** (1 - T) + 310_000 ** (1 - T) = k
+                    #
+                    # Solving for d_z, we get the following calculation:
+                    #   d_z = 100_000 - (k - 310_000 ** (1 - T)) ** (1 / (1 - T)) = 9740.77011591768
+                    #
+                    # The output is d_x = c * d_z. Since c = 1, d_x = d_z.
+                    # Note that this is slightly smaller than the without slippage value
+                    without_fee=9740.77011591768,
+                    # Combining the without_fee and the fee, we calculate with_fee as:
+                    #   with_fee = 9740.77011591768 - 2.4454185805248496 = 9738.324697337155
+                    with_fee=9738.324697337155,
+                ),
+            ),
+            # TODO: The slippage should arguably be much higher. This is something
```

Comment:
This is a repeat of the above comment, right? I recommend just including one TODO comment.